pygfx/renderers/wgpu/_pipeline.py

```diff
@@ -497,8 +497,10 @@ def update_shader_data(self, wobject, changed):
             changed.update(("bindings", "pipeline_info", "render_info"))
 
         if "bindings" in changed:
+            self.shader.unlock_hash()
             with wobject.tracker.track_usage("!bindings"):
                 self.bindings_dicts = self.shader.get_bindings(wobject, self.shared)
+            self.shader.lock_hash()
```

Comment:
This looks like something that needs a try/finally?

Reply:
No, if an exception is raised in `get_bindings()` then the code does not proceed anyway, so there is no need to apply the lock.